src/Functions/generateSerialID.cpp

```diff
@@ -126,7 +138,10 @@ class FunctionSerial : public IFunction
                     name, max_series);
         }
         else
-            keeper->createIfNotExists(serial_path, "0");
+        {
+            String initial_value = has_start_value ? toString(start_value) : "0";
```

Comment:
I think it worth the readability.
```suggestion
            const auto initial_value = toString(start_value);
```